src/optimizer/loadOptimizers.ts

```diff
@@ -49,9 +56,29 @@ export async function loadOptimizers(
           edges.push([opt.name, preceedingOptimizer.name]);
         });
       } else {
-        logger.error(
-          `The ${opt.name} optimizer specifies an unknown optimizer in runAfter: ${dependsOn}`
-        );
+        // the preceeding optimizer(s) may exist, but not be enabled
+        // this is still an error, but needs a different message
+        const offPrecedingOptimizers = offOptimizers
+          .filter(o =>
+            o.name !== opt.name && dependsOn instanceof RegExp
+              ? dependsOn.test(o.name)
+              : dependsOn === o.name
+          )
+          .map(o => o.name);
+        if (offPrecedingOptimizers.length > 0) {
+          const plural = offPrecedingOptimizers.length > 1;
+          logger.error(
+            `The ${opt.name} optimizer specifies ${
+              plural ? 'optimizers' : 'an optimizer'
+            } in runAfter that ${plural ? 'are' : 'is'} not enabled: ${offPrecedingOptimizers.join(
+              ', '
+            )}`
+          );
```

Comment:
Since this is in an `else` block, it will only be run if the `runAfter` value had zero matches.  What if a regular expression was used and some active optimizers matched but some disabled optimizers also matched?  On one hand, it feels like that's potentially still an issue and we want to log an error.  On the other hand, we wouldn't want it to trigger an error on an optimizer that basically says "run me last" because it doesn't really care what came before it; only that it runs last.  So... maybe this is fine as-is?

Or maybe we should consider if this is really an error condition at all?  Sometimes we say "runAfter" because there is a true dependency on the previous optimizer (if it doesn't run, we are toast), but sometimes we might say "runAfter" because we know that if the previous optimizer will be run, it might affect our output, so it's best to run it first (but if it's not run, that's not a big deal). So... maybe we're being too strict?  I'm honestly not sure.

Reply:
This is a good point. I had always interpreted `runAfter` as saying "these are the necessary preconditions in order to correctly run," but that's just my interpretation, and that doesn't allow for differentiating between the cases you pointed out. In order to be able to represent those ideas, we might need something like `tryRunAfter` for things that are okay to turn off and `mustRunAfter` for things that are not okay to turn off. So something that just wants to get run after other things, but doesn't have any hard preconditions, can have a `tryRunAfter` that will match most other things.

The more I think about your second paragraph here, the more I like that idea. That is, there shouldn't be a case where running optimizer A, but not optimizer B, results in incorrect output. If that statement is correct, then it shouldn't be necessary to check for these errors at all. I'll take a look at our current set of optimizers and see if there are any that break this rule. If nothing breaks this rule, then I think a slight change to this implementation and a removal of the error case is fine.